sdk/monitor/Azure.Monitor.OpenTelemetry.LiveMetrics/src/LiveMetricsExporter.cs

```diff
@@ -24,14 +33,70 @@ public LiveMetricsExporter(DoubleBuffer doubleBuffer, LiveMetricsExporterOptions
 
         public override ExportResult Export(in Batch<Metric> batch)
         {
-            MonitoringDataPoint monitoringDataPoint = new MonitoringDataPoint();
             DocumentBuffer filledBuffer = _doubleBuffer.FlipDocumentBuffers();
+            MonitoringDataPoint monitoringDataPoint = new()
+            {
+                Version = SdkVersionUtils.s_sdkVersion.Truncate(SchemaConstants.Tags_AiInternalSdkVersion_MaxLength),
+                InvariantVersion = CurrentInvariantVersion,
+                Instance = MetricResource?.RoleInstance,
+                RoleName = MetricResource?.RoleName,
+                MachineName = Environment.MachineName,
+                StreamId = GetStreamId(),
+                Timestamp = DateTime.UtcNow,
+                //TODO: Provide feedback to service team to get this removed, it not a part of AI SDK.
+                // TransmissionTime = DateTime.UtcNow,
+                IsWebApp = IsWebAppRunningInAzure(),
+                PerformanceCollectionSupported = false,
+                // AI SDK relies on PerformanceCounter to collect CPU and Memory metrics.
+                // Follow up with service team to get this removed for OTEL based live metrics.
+                // TopCpuProcesses = null,
+                // TODO: Configuration errors are thrown when filter is applied.
+                // CollectionConfigurationErrors = null,
+            };
 
             foreach (var item in filledBuffer.ReadAllAndClear())
             {
                 monitoringDataPoint.Documents.Add(item);
             }
 
+            foreach (var metric in batch)
+            {
+                if (!LiveMetricsExtractionProcessor.s_liveMetricNameMapping.TryGetValue(metric.Name, out var metricName))
+                {
+                    continue;
+                }
+
+                var liveMetricPoint = new Models.MetricPoint
+                {
+                    Name = metricName
+                };
+
+                foreach (ref readonly var metricPoint in metric.GetMetricPoints())
```

Comment:
Just wanted to make one callout.
If this saves on allocations, then excellent :)

Reusing the liveMetricPoint might cause problems if the Add at line 94 is by ref. Should test to confirm that we're adding the full collection of metricpoints, or just several copies of the final metricpoint.

Reply:
It should not have any impact, here is a similar implementation - https://github.com/Azure/azure-sdk-for-net/blob/c284046de29a1561373843a0f4f33600c474348a/sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/MetricHelper.cs#L23

I agree, all of our implementation should be supported with test. Once we could get PoC working, we should add tests  on priority before adding new implementations.

Reply:
I think this won't work. We are overriding the same metricPoint. Am I missing something?